### PR TITLE
Runtime: Add support for Builtin.Int512

### PR DESCRIPTION
**Pull request by aschwaighofer**
rdar://31540879

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

File: stdlib/public/runtime/KnownMetadata.cpp [hunk 1]
```diff
@@ -57,6 +60,8 @@ const ValueWitnessTable swift::VALUE_WITNESS_SYM(Bi128_) =
   ValueWitnessTableForBox<NativeBox<int128_like, 16>>::table;
 const ValueWitnessTable swift::VALUE_WITNESS_SYM(Bi256_) =
   ValueWitnessTableForBox<NativeBox<int256_like, 32>>::table;
+const ValueWitnessTable swift::VALUE_WITNESS_SYM(Bi512_) =
+  ValueWitnessTableForBox<NativeBox<int512_like, 32>>::table;
```
Comment (stephentyrone) [inline stdlib/public/runtime/KnownMetadata.cpp]: Should this be `, 64>>`?

Comment (aschwaighofer) [inline stdlib/public/runtime/KnownMetadata.cpp]: Yes. Thanks!

**Comment by aschwaighofer** [comment]
@swift-ci Please smoke test

**Comment by stephentyrone** [comment]
LGTM, thanks!

**Comment by dabrahams** [comment]
A couple questions:

1. What prompted this?
2. We should do a perf comparison with `DoubleWidth<DoubleWidth<Int64>>` :-)

**Comment by aschwaighofer** [comment]
This was needed after adding 64B SIMD element types.

**Pull request merged**